### PR TITLE
Theme element dual inheritance

### PR DESCRIPTION
**Pull request by teunbrand**
This PR aims to fix a problem I ran into during 2 previous PRs (#5167 and [#4879](https://github.com/tidyverse/ggplot2/pull/4879#issuecomment-1369235237)). Instead of the pragmatic hotfixes in those PRs, this one addresses the root cause.

The problem is that `calc_element()` could not be used to calculate any part of the theme that had dual `<character/numeric>` inheritance. Under the hood, an exception was made in the validation to allow character elements to also be numeric. This was the case for `legend.position` that could be either `"top"` or `c(0, 1)`, for example. This is all fine, however, `calc_element()` did not special-case the `<character/numeric>` dual inheritance, and would throw an error if you provide a numeric vector (see reprex).

The solution in this PR is to properly give those elements that should support dual inheritance a dual class. By doing this, we no longer need to special-case in the validator, and `calc_element()` gives the right answers. Also, we can let `aspect.ratio` inherit from `<numeric>` now, whereas previously it would throw uninformative errors from later in the plotting process.

A demonstration of before this PR (backtraces omitted):

``` r
library(ggplot2)

theme <- theme_get() +
  theme(
    aspect.ratio = "foobar",
    plot.tag.position = c(0, 1)
  )

calc_element("aspect.ratio", theme)
#> [1] "foobar"
calc_element("plot.tag.position", theme)
#> Error:
#> ! Theme element `plot.tag.position` must have class <character>
```

Whereas now:

``` r
devtools::load_all("~/packages/ggplot2")
#> ℹ Loading ggplot2

calc_element("aspect.ratio", theme)
#> Error:
#> ! Theme element `aspect.ratio` must have class <numeric>

calc_element("plot.tag.position", theme)
#> [1] 0 1
```

<sup>Created on 2023-02-02 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

File: R/theme-elements.r [hunk 1]
```diff
@@ -476,12 +476,12 @@ el_def <- function(class = NULL, inherit = NULL, description = NULL) {
   legend.key.height   = el_def("unit", "legend.key.size"),
   legend.key.width    = el_def("unit", "legend.key.size"),
   legend.text         = el_def("element_text", "text"),
-  legend.text.align   = el_def("character"),
+  legend.text.align   = el_def("numeric"),
```
[the remark this answers is not in the document]
Comment (teunbrand) [reply R/theme-elements.r:479]: Currently, giving a `character(1)` results in a grid-error, i.e. `legend.text.align = "left"` gets:

> `Error in valid.charjust(just) : invalid vertical justification`

Whereas with this PR, the error will be:

> `! The `legend.text.align` theme element must be a <numeric> object.`

Which is slightly more informative. I can see how allowing character justification might help here, but that might be a separate PR from this one.

[the remark this answers is not in the document]
Comment (thomasp85) [reply R/theme-elements.r:479]: huh... surprising - but agreed that it is separate from this

**Review by thomasp85 — APPROVED** [review]
LGTM

**Comment by teunbrand** [comment]
Thanks for the review!

**Pull request merged**